builders/integrations/wallets/rainbowkit.md

```diff
@@ -0,0 +1,328 @@
+
+=== "npm"
+
+    ```bash
+    npm run dev
+    ```
+
+=== "pnpm"
+
+    ```bash
+    pnpm run dev
+    ```
+
+=== "yarn"
+
+    ```bash
+    yarn dev
+    ```
+
+If you navigate to [http://localhost:3000](http://localhost:3000){target=\_blank},  you should see the starter Next.js application in your browser.
+
+To test the RainbowKit connection, you will use the MetaMask mobile app. To follow this guide, you must have established a connection to the Moonbase Alpha TestNet on the MetaMask mobile app.
+
+You can connect your MetaMask mobile wallet to the Moonbase Alpha TestNet in a couple of ways. You can manually add the Moonbase Alpha TestNet configurations from the **Networks** section of the **Settings** menu. Or you can also open up the **Browser** from MetaMask mobile and navigate to the [Moonbeam documentation site (docs.moonbeam.network)](/){target=\_blank}, click on **Connect MetaMask** at the top of the page, and select **Moonbase Alpha** from the menu. Follow the prompts to automatically add Moonbase Alpha as a custom network.
+
+Next, safely add your `projectId` to your application: 
+
+- Create a `.env.local` file in the root directory of your project
+
+```bash
+touch .env.local
+```
+
+- Add your `projectId` to this file
+
+```text title=".env.local"
+NEXT_PUBLIC_PROJECT_ID='INSERT_PROJECT_ID'
+```
+
+- Locate your `.gitignore` file in this same directory and ensure `.env*.local` is included in the list of files to ignore. This will prevent committing your `projectId` to GitHub
+
+In the next section, you will use this stored `projectId` when setting up the `wagmi` config.
+
+### Connect DApp to MetaMask Mobile {: #connect-dapp-to-metamask-mobile }
+
+In the next sections, you will complete the steps needed to use RainbowKit to connect your dApp to MetaMask's mobile wallet:
+
+1. Import RainbowKit, Wagmi, and TanStack Query
+2. Setup configuration for Wagmi
+3. Wrap your application with providers
+4. Add the connect button
+
+### Import RainbowKit, Wagmi, and TanStack Query
+
+Ensure you are still in your project's root directory, then create a new file called `wagmi.ts`. This file will contain the imports and configuration needed to connect your dApp to mobile wallets and interact with blockchains. 
+
+```bash
+touch wagmi.ts
+```
+
+Start by adding the imports for RainbowKit, Wagmi, and TanStack Query. You will also import your desired list of supported chains from `wagmi/chains`.  Wagmi provides an extensive list of [supported chains](https://wagmi.sh/core/api/chains#:~:text=core/chains%27-,Available,-Chains){target=\_blank}. The expected names for Moonbeam networks are as follows:
+
+=== "Moonbeam"
+
+    ```js
+    `moonbeam`
+    ```
+
+=== "Moonriver"
+
+    ```js
+    `moonriver`
+    ```
+
+=== "Moonbase Alpha"
+
+    ```js
+    `moonbaseAlpha`
+    ```
+
+Add the `wagmi/chains` import with `moonbeam`, `moonriver`, and `moonbaseAlpha` as the supported chains. 
+
+```ts
```

Comment:
can you please add the file as the title here (like you did above) and also make sure the other snippets have a title?